test/modules/CloudToDeviceMessageTester/Receiver.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace CloudToDeviceMessageTester
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil.TestResults;
+    using Microsoft.Extensions.Logging;
+    using TransportType = Microsoft.Azure.Devices.TransportType;
+
+    sealed class Receiver : CloudToDeviceMessageTesterBase
+    {
+        DeviceClient deviceClient;
+
+        public Receiver(
+            ILogger logger,
+            string iotHubConnectionString,
+            string deviceId,
+            string moduleId,
+            TransportType transportType,
+            TimeSpan testDuration,
+            TestResultReportingClient testResultReportingClient)
+            : base(logger, iotHubConnectionString, deviceId, moduleId, transportType, testDuration, testResultReportingClient)
+        {
+        }
+
+        public override void Dispose() => this.deviceClient?.Dispose();
+
+        public async Task ReportTestResult(Message message)
+        {
+            string trackingId = message.Properties[TestConstants.Message.TrackingIdPropertyName];
```

Comment:
think about if this can be shared with existing message scenario; do it in another PR.